src/file-upload/file-upload.tsx

```diff
@@ -65,23 +99,142 @@ export const FileUpload = ({
         }
     };
 
+    const handleDescriptionUpdate =
+        (item: FileItemProps) => (description: string) => {
+            if (onEdit) {
+                const updatedItem = { ...item, description };
+                onEdit(updatedItem);
+            }
+        };
+
+    const handleInitiateEdit = (item: FileItemProps) => () => {
+        setItemsRenderMode((prevState) => {
+            return prevState.map((renderItem) => {
+                if (isEqual(item, renderItem.fileItem)) {
+                    return {
+                        ...renderItem,
+                        renderMode: "edit",
+                    };
+                }
+
+                return renderItem;
+            });
+        });
+    };
+
+    const handleCancel = (item: FileItemProps) => () => {
+        setItemsRenderMode((prevState) => {
+            return prevState.map((renderItem) => {
+                if (isEqual(item, renderItem.fileItem)) {
+                    return {
+                        ...renderItem,
+                        renderMode: "default",
+                    };
+                }
+
+                return renderItem;
+            });
+        });
+    };
+
     // =========================================================================
-    // RENDER FUNCTIONS
+    // HELPER FUNCTIONS
     // =========================================================================
-    const renderItems = () => {
-        if (!fileItems || fileItems.length === 0) return null;
+    const checkEditable = (type: string) => {
+        return editableFileItems && FileUploadHelper.isSupportedImageType(type);
+    };
+
+    const getItemsRenderMode = (
+        fileItems: FileItemProps[]
+    ): FileItemRenderProps[] => {
+        if (!fileItems || fileItems.length === 0) return [];
+
+        return fileItems.map((item) => {
+            return {
+                fileItem: item,
+                renderMode:
+                    FileUploadHelper.isSupportedImageType(item.type) &&
+                    item.description === undefined
+                        ? "edit"
+                        : "default",
+            };
+        });
+    };
 
+    const getArrangedItemsToRender = (
```

Comment:
what was the consideration for nesting editable items in an array and rendering it within another wrapper container?

Reply:
There was a ui requirement that I could not achieve with css child flags, and find it cumbersome to use props. Grouping allowed me to do that